consensus/state_processing/src/per_block_processing/process_operations.rs

```diff
@@ -658,11 +675,73 @@ pub fn process_consolidation_requests<E: EthSpec>(
     Ok(())
 }
 
+fn is_valid_switch_to_compounding_request<E: EthSpec>(
+    state: &BeaconState<E>,
+    consolidation_request: &ConsolidationRequest,
+    spec: &ChainSpec,
+) -> Result<bool, Error> {
+    // Switch to compounding requires source and target be equal
+    if consolidation_request.source_pubkey != consolidation_request.target_pubkey {
+        return Ok(false);
+    }
+
+    // Verify pubkey exists
+    let Some(source_index) = state
+        .pubkey_cache()
+        .get(&consolidation_request.source_pubkey)
+    else {
+        // source validator doesn't exist
+        return Ok(false);
+    };
+
+    let source_validator = state.get_validator(source_index)?;
+    // Verify the source withdrawal credentials
+    if let Some(withdrawal_address) = source_validator.get_eth1_withdrawal_credential(spec) {
+        if withdrawal_address != consolidation_request.source_address {
+            return Ok(false);
+        }
+    } else {
+        // Source doen't have execution withdrawal credentials
```

Comment:
```suggestion
        // Source doesn't have eth1 withdrawal credentials
```
the distinction here is important

Reply:
Fixed